llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp

```diff
@@ -6224,3 +6224,89 @@ void CombinerHelper::applyCommuteBinOpOperands(MachineInstr &MI) {
   MI.getOperand(2).setReg(LHSReg);
   Observer.changedInstr(MI);
 }
+
+// Transform build_vector of binop(_,C) -> binop(BV, constant BV).
+bool CombinerHelper::matchBuildVectorToBinOp(MachineInstr &MI,
+                                             BuildFnTy &MatchInfo) {
+  GBuildVector *BuildVector = cast<GBuildVector>(&MI);
+  Register Dst = BuildVector->getReg(0);
+
+  unsigned NumOfSources = BuildVector->getNumSources();
+  if (NumOfSources == 1)
```

Comment:
I don't think this is legal?

Reply:
The original RISC-V code contains this check.

Reply:
It was unnecessary. We don't have 1x vectors in LLT and the verifier implies it's not possible

Reply:
It was probably a profitability check. Removed.